bindings/py/src/plugin/RegisteredRegionImplPy.hpp

```diff
@@ -76,43 +91,72 @@ namespace nupic
   public:
 	  RegisteredRegionImplPy(const std::string& classname, const std::string& module="")
 			: RegisteredRegionImpl(classname, module) {
-		if (python_region_count == 0) {
+		if (python_node_count <= 0) {
```

Comment:
> RegisteredRegionImplPy.hpp line 94
`if (python_node_count <= 0) {`

Well, 'python_node_count' is a global.  (I hate globals but nothing else seemed to let me create a singleton without a lot of mess).  I was protecting myself for in the very unlikely event that we might somehow decrement more than once.  Since the decrement is in the distructor, it is probably not going to happen.  But if it should be less than 0, it self corrects on next registration.

Reply:
I was worried about this line 
https://github.com/htm-community/nupic.cpp/pull/147/files/386af995fac6400ef05a6aa1baaae0d153183c83#diff-0abfb513260060b21797b16673bf7460R104 
(could add the NTA_ASSERT >= 0 just above it, more to make clear the neaning than anything else) 

...but this line saves it: 
https://github.com/htm-community/nupic.cpp/pull/147/files/386af995fac6400ef05a6aa1baaae0d153183c83#diff-0abfb513260060b21797b16673bf7460R102 

But likely here you want to be more proactive to destroy: 

https://github.com/htm-community/nupic.cpp/pull/147/files/386af995fac6400ef05a6aa1baaae0d153183c83#diff-0abfb513260060b21797b16673bf7460R109